### PR TITLE
add paused flag to ferry and elevator

### LINKED ISSUE #342: Eleveator paused flag

### PR DESCRIPTION
**Pull request by aborilov**
**What this PR does / why we need it**:
Allow pausing reconciliation of operator managed instances.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #342, #340, #338, #343 

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/operator/v1alpha1/catapult_types.go [hunk 1]
```diff
@@ -40,19 +40,6 @@ type CatapultSpec struct {
 	Paused PausedFlagType `json:"paused,omitempty"`
 }
 
-// PausedFlagType represents a enable/disable flag
```
Comment (nmiculinic) [inline pkg/apis/operator/v1alpha1/catapult_types.go:43]: why is this removed? This PR, at least by the title, has nothing to do with the catapult.

File: pkg/apis/operator/v1alpha1/elevator_types.go [hunk 2]
```diff
@@ -50,6 +52,7 @@ type ElevatorPhaseType string
 // Values of ElevatorPhaseType.
 const (
 	ElevatorPhaseReady       ElevatorPhaseType = "Ready"
+	ElevatorPhasePaused      ElevatorPhaseType = "Paused"
```
Comment (nmiculinic) [inline pkg/apis/operator/v1alpha1/elevator_types.go:55]: when is the high time for autogenerating this stuff? we are doing it manually every time, both in kubecarrier, and in bulward.

File: pkg/apis/operator/v1alpha1/elevator_types.go [hunk 3]
```diff
@@ -211,6 +219,61 @@ func (s *Elevator) SetUnReadyCondition() bool {
...
+	for _, condition := range s.Status.Conditions {
+		if condition.Type == ElevatorPaused &&
+			condition.Status == ConditionTrue {
+			return true
+		}
+	}
+	return false
+}
+
+func (s *Elevator) SetPausedCondition() bool {
```
Comment (nmiculinic) [inline pkg/apis/operator/v1alpha1/elevator_types.go:237]:
```suggestion
func (s *Elevator) Pause() bool {
```

File: pkg/apis/operator/v1alpha1/elevator_types.go [hunk 4]
```diff
@@ -211,6 +219,61 @@ func (s *Elevator) SetUnReadyCondition() bool {
...
+			Type:    ElevatorPaused,
+			Status:  ConditionTrue,
+			Reason:  "Paused",
+			Message: "Reconcilation is paused",
+		})
+	}
+	return changed
+}
+
+func (s *Elevator) SetUnPausedCondition() bool {
```
Comment (nmiculinic) [inline pkg/apis/operator/v1alpha1/elevator_types.go:263]:
```suggestion
func (s *Elevator) Unpause() bool {
```

File: pkg/apis/operator/v1alpha1/catapult_types.go (hunk 1 above)
Comment (aborilov) [reply pkg/apis/operator/v1alpha1/catapult_types.go:43]: it's a common type for all operators, it was in `catapult` only because was implemented first for `catapult` but in this PR it was moved to `types.go`

File: pkg/apis/operator/v1alpha1/elevator_types.go (hunk 2 above)
Comment (aborilov) [reply pkg/apis/operator/v1alpha1/elevator_types.go:55]: let's add separate issue if we don't have one already and finally do it( just do it :))

File: pkg/apis/operator/v1alpha1/elevator_types.go (hunk 3 above)
Comment (aborilov) [reply pkg/apis/operator/v1alpha1/elevator_types.go:237]: let's leave it consistent with `SetReadyCondition` and `SetTerminatingCondition`

File: pkg/apis/operator/v1alpha1/elevator_types.go (hunk 4 above)
Comment (aborilov) [reply pkg/apis/operator/v1alpha1/elevator_types.go:263]: let's leave it consistent with `SetUnReadyCondition`
